src/docs/spec-formats-rdf.md

```diff
@@ -0,0 +1,366 @@
+format is not required, implementations can freely decide whether to
+serialise identifiers as IRIs or CURIEs (assuming the concrete RDF
+syntax allows that of course).
+
+### Extension definitions
+Extension definitions MAY be omitted if the RDF file is only intended to
+be used by RDF applications.
+
+Conversely, they SHOULD be included if the set is intended to be
+convertible back to another SSSOM format.
+
+> Non-normative explanation
+>
+> The whole point of an extension definition in SSSOM is to provide (1)
+> a property that confers some meaning to the extension, and (2) the
+> type of the expected values. In RDF, as described
+> [above](#extension-slots), those two bits of information are already
+> contained in the triple that represent the extension slot, so there is
+> no need for an additional definition.
+>
+> But the extension definition also provides the `slot_name` which is
+> used to represent the extension slot in other formats (especially
+> SSSOM/TSV), so if conversion back to other SSSOM formats is required,
+> ensuring that the extension definitions are present in the RDF
+> serialisation is helpful.
+
+### Propagation and condensation
+Propagatable slots can be represented in RDF indifferently in their
+propagated or condensed form, following the [normal
+rules](spec-model.md##propagation-of-mapping-set-slots) for propagation
+and condensation.
+
+But if the RDF file is intended to be used by generic, non-SSSOM-aware
+RDF applications, then implementations SHOULD serialise propagatable
+slots in their propagated form.
+
+> Non-normative explanation
+>
+> Propagation is a SSSOM-specific concept. If a RDF application is
+> provided with a RDF file representing a set with condensed slots, the
+> application will not know to propagate the condensed slots at the set
+> level down to the level of the individual mappings, which will result
+> in the application having an incomplete view of the mappings.
+
+
+## Compatibility with pre-standard RDF representations
+The present specification of the SSSOM/RDF format differs slightly from
+what several implementations of SSSOM have been producing before the
+format was formally specified.
+
+In the name of backward compatibility, implementations MAY support the
+alternative rules described in the following subsections when
+deserialising from RDF.
+
+Implementations MUST NOT follow these rules when serialising to RDF.
+
+### Representation of slots typed as `sssom:NonRelativeURI`
+Implementations MAY accept a value represented as a `xsd:anyURI`
+literal.
+
+### Representation of slots typed as an enumeration
+Implementations MAY accept a value represented as a string literal, even
+if the value is defined in the LinkML model as having an associated
+`meaning` property.
+
+For example, implementations MAY accept
+
+```ttl
+?mapping sssom:predicate_modifier "Not"^^xsd:string .
+```
+
+as an alternative to
+
+```ttl
+?mapping sssom:predicate_modifier sssom:NegatedPredicate .
```

Comment:
why not just decide on one and say "this is the standard"?

Reply:
That’s what did. This:

```ttl
?mapping sssom:predicate_modifier sssom:NegatedPredicate .
```

**is** the standard.

But the decision to standardize that form has only been made a few weeks ago. Before that, both SSSOM-Java and SSSOM-Py have been producing the string literal form (SSSOM-Java for the past 8 months – since version 1.1, which introduced RDF support – and SSSOM-Py for as long as it has existed). In fact SSSOM-Py _still_ produces the string literal form to this day.

So for backwards compatibility (which is the entire point of this section, “compatibility with pre-standard representations”), implementations MAY support the old string literal form, even though it is _not_ the standard form.

Reply:
Makes sense.